### PR TITLE
WIP : adding american format syntax

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**

### CODE REVIEW

File: format_test.go [hunk 1]
```diff
@@ -30,13 +30,13 @@ var formatTests = []struct {
...
 		s: pointer(0),
 	}, "", 1569585600},
-	{"timestamp", formatsMap["timestamp"], &result{
-		y: pointer(2019),
-		h: pointer(8),
-		m: pointer(8),
-		i: pointer(30),
-		d: pointer(27),
-		s: pointer(0),
+	// {"timestamp", formatsMap["timestamp"], &result{
```
Comment (M-ZubairAhmed) [inline format_test.go]: temporarily disabled due to https://github.com/carmo-evan/strtotime/issues/4

File: format_test.go [hunk 2]
```diff
@@ -52,6 +52,11 @@ var formatTests = []struct {
 		i: pointer(30),
 		s: pointer(0),
 	}, "last", 1551342600},
+	{"american", formatsMap["american"], &result{
```
Comment (M-ZubairAhmed) [inline format_test.go]: i am unable to write a correct test for this format, could you please help me explain it out

File: format_test.go [hunk 3]
```diff
@@ -52,6 +52,11 @@ var formatTests = []struct {
...
+		y: pointer(2019),
+		m: pointer(10),
```
Comment (carmo-evan) [inline format_test.go]: Months are zero-indexed, so they range from 0 (January) to 11 (December). If you change this to 9 it should work.
```suggestion
		m: pointer(9),
```

File: strtotime_test.go [hunk 4]
```diff
@@ -33,6 +33,7 @@ var parseTests = []struct {
 	{"tomorrow 01am", time.Date(now.Year(), now.Month(), now.Day()+1, 1, 0, 0, 0, time.UTC).Unix(), true},
 	{"last day of October 1am", time.Date(now.Year(), time.October, 31, 1, 0, 0, 0, time.UTC).Unix(), true},
 	{"1am 2pm", 0, false},
+	{"09/27/19", time.Date(2019, time.September, 27, 0, 0, 0, 0, time.UTC).Unix(), true},
```
Comment (M-ZubairAhmed) [inline strtotime_test.go]: Tests keep failing showing the following error,
```bash
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4fa5ec]

goroutine 34 [running]:
testing.tRunner.func1(0xc000145e00)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x529300, 0x644e20)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
strtotime.(*result).ymd(0xc0001d14a0, 0x7e3, 0x8, 0x1b, 0x0, 0x0)
	/home/zubair/projects/forked/strtotime-go/result.go:55 +0x8c
strtotime.formats.func13(0xc0001d14a0, 0xc0001e2810, 0x3, 0x3, 0x4, 0x4)
	/home/zubair/projects/forked/strtotime-go/format.go:280 +0x129
strtotime.Parse(0x5549f0, 0x8, 0x4bda30, 0x60b748, 0x63a460, 0x0, 0x1)
	/home/zubair/projects/forked/strtotime-go/strtotime.go:27 +0x18a
strtotime.TestParse.func1(0xc000145e00)
	/home/zubair/projects/forked/strtotime-go/strtotime_test.go:42 +0x4d
testing.tRunner(0xc000145e00, 0xc0001d8490)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
exit status 2
FAIL	strtotime	0.009s

```

**Pull request merged**